src/request_id.rs

```diff
@@ -25,12 +25,24 @@ use uuid::Uuid;
 ///   format!("{}", uuid)
 /// }
 /// ```
+///
+/// Optionally, using the `uuid_v7` feature flag will allow [`RequestId`] to use UUID v7 instead of the currently used UUID v4.
+///
+/// However, the [`uuid`] crate requires a compile time flag `uuid_unstable` to be passed in `RUSTFLAGS="--cfg uuid_unstable"` in order to compile. You can read more about it [here](https://docs.rs/uuid/latest/uuid/#unstable-features).
+///
 #[derive(Clone, Copy, Debug)]
 pub struct RequestId(Uuid);
 
 impl RequestId {
     pub(crate) fn generate() -> Self {
-        Self(Uuid::new_v4())
+        #[cfg(not(feature = "uuid_v7"))]
+        {
+            Self(Uuid::new_v4())
+        }
+        #[cfg(feature = "uuid_v7")]
+        {
+            Self(Uuid::now_v7())
+        }
```

Comment:
Considering that `--cfg uuid_unstable` must be passed, I think we should check both the `uuid_v7` and the cfg itself, defaulting to V4 if either hasn't been set properly.

Reply:
I.e. `#[cfg(all(uuid_unstable, feature = "uuid_v7"))]`

Reply:
I was of a mind that in the future if the crate removes this requirement of having the cfg flag we would need a code change, but regardless, I agree that cascading the dependency for uuid crate to here will definitely provide more visibility.